src/ECModel.jl

```diff
@@ -852,6 +853,187 @@ function split_financial_terms(ECModel::AbstractEC, profit_distribution=nothing)
+        user_set = get_user_set(ECModel)
+        profit_distribution = JuMP.Containers.DenseAxisArray(
+            [ 0.0
+                for y in year_set, u in setdiff(user_set_financial, [EC_CODE])
+            ]
+            , year_set, user_set,
+        )
+    end
+    @assert termination_status(ECModel) != MOI.OPTIMIZE_NOT_CALLED
+
+    user_set = axes(profit_distribution)[2]
+    ann_factor = [1. ./((1 + field(gen_data, "d_rate")).^y) for y in year_set]
+
+    # Investment costs
+    CAPEX = JuMP.Containers.DenseAxisArray(
+        [(y==1) ? sum(Float64[get_value(ECModel.results[:CAPEX_tot_us], u)]) : 0.0
+            for y in year_set, u in setdiff(user_set_financial, [EC_CODE])
+            ]
+           , year_set, user_set
+    )
+    # Maintenance costs
+    Ann_Maintenance = JuMP.Containers.DenseAxisArray(
+        [get_value(ECModel.results[:C_OEM_tot_us], u)
+            for y in year_set, u in setdiff(user_set_financial, [EC_CODE])
+        ]
+        , year_set, user_set
+    )
+    # Replacement costs
+    Ann_Replacement = JuMP.Containers.DenseAxisArray(
+            [get_value(ECModel.results[:C_REP_tot_us][y, :], u)
+                for y in year_set, u in setdiff(user_set_financial, [EC_CODE])
+            ]
+            , year_set, user_set
+     )
+    # Recovery value
+    Ann_Recovery = JuMP.Containers.DenseAxisArray(
+        [get_value(ECModel.results[:R_RV_tot_us][y, :], u)
+            for y in year_set, u in setdiff(user_set_financial, [EC_CODE])
+            ]
+                , year_set, user_set
+    )
+
+    # Peak energy charges
+    Ann_peak_charges = JuMP.Containers.DenseAxisArray(
+        [get_value(ECModel.results[:C_Peak_tot_us], u)
+            for y in year_set, u in setdiff(user_set_financial, [EC_CODE])
+            ]
+                , year_set, user_set
+    )
+
+    # Get revenes by selling energy and costs by buying or consuming energy
+    zero_if_negative = x->((x>=0) ? x : 0.0)
+    Ann_energy_revenues = JuMP.Containers.DenseAxisArray(
+        [(u in axes(ECModel.results[:R_Energy_us])[1]) ? sum(zero_if_negative.(ECModel.results[:R_Energy_us][u,:])) : 0.0
+            for y in year_set, u in setdiff(user_set_financial, [EC_CODE])
+            ]
+            , year_set, user_set
+    )
+    Ann_energy_costs = JuMP.Containers.DenseAxisArray(
+        [(u in axes(ECModel.results[:R_Energy_us])[1]) ? sum(zero_if_negative.(.-(ECModel.results[:R_Energy_us][u, :]))) : 0.0
+        for y in year_set, u in setdiff(user_set_financial, [EC_CODE])
+            ]
+            , year_set, user_set
+    )
+    Ann_net_energy_costs = Ann_energy_costs .- Ann_energy_revenues
+    
+    # Total OPEX costs
+    OPEX = Ann_Maintenance .+ Ann_peak_charges .+ Ann_net_energy_costs
+
+    # get NPV given the reward allocation
+    #Check how to proceed
+    NPV = profit_distribution
+
+    # Total reward
+    Ann_reward = NPV .+ CAPEX .+ OPEX .+ Ann_Replacement .- Ann_Recovery

```

Comment:
So, the annualized reward you should annualize CAPEX, OPEX, Ann_Replacement and Ann_Recovery for all users, so to have a DenseAxisArray with only 1 dimension: the user_list.
Then you can do the sum/differences as done here. The vector that you get is the total annualized reward and by dividing for the sum of the ann_factor, you can get the reward allocation for every year.

The NPV may be a single dense axis array that is the NPV by user/agg

Reply:
The proposed approach unfortunately does not split the reward by user.
The goal of the original code was to derive what reward was allocated for every user and year, according to the NPV calculated with (game-theoretic) allocations or alternative techniques.
The current implementation unfortunately does not capture that.

Something like the following may work:
```julia
PVTotalLessReward = DenseAxisArray(
(CAPEX .+ OPEX .+ Ann_Replacement .- Ann_Recovery).data * ann_factor,
users_set_financial,
)
YearlyReward = (NPV .- PVTotalLessReward)./(sum(ann_factor)
```

Where YearlyReward is the yearly reward by user that is constant every year.
This can then be used to generate the 2D dense axis array equivalent to CAPEX/OPEX where we specify the reward by user and by year. The reward is constant by year, with the exception of y=0

Reply:
Let me know where you don't get the equations; unfortunately I'm at a conference and I'm a bit delayed